lib/segment/src/index/hnsw_index/graph_links/serializer.rs

```diff
@@ -99,6 +99,9 @@ pub fn serialize_graph_links<W: Write + Seek>(
     let mut offset = 0; // elements for Plain, bytes for Compressed/CompressedWithVectors
     let mut offsets = Vec::with_capacity(total_offsets_len as usize);
     offsets.push(0);
+
+    #[expect(clippy::needless_range_loop)]
+    // this clippy lint is positively demented, can't wait till they remove it 🙄
```

Comment:
The most braindead diagnostic that I've seen for a while. I think it will be reworked or removed soon, but maybe we should just disable it globally.